dataset/create_twitter_dataset.py

```diff
@@ -0,0 +1,199 @@
+# This file takes the twitter dataset and transform it into the
+# format described in https://github.com/paivaspol/EECS591/issues/25
+# The result dataset will support data interdependency.
+import argparse
+import math
+import random
+import uuid
+
+# Constants for twitter modes.
+TWITTER_RT = 'RT'
+TWITTER_MT = 'MT'
+TWITTER_RE = 'RE'
+UUID_LIST_FILE = 'uuid_list.txt'
+USER_IP_MAP_FILE = 'user_ip_map.txt'
+TWEET_SIZE = 140
+TIMELINE_LIMIT = 20
+LOG_FORMAT = '%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n'  # Schema: [timestamp]\t[target_uuid]\t[src]\t[source_dependency]\t[dest]\t[type]\t[status]\t[size]
+
+# Generates the dataset.
+#
+# params:
+#   filename: the name of the file containing the twitter data
+#   user_ip_map: a map containing the user
+def generate_dataset(filename, user_ip_map, uuid_server_map, rt_re_set):
+    print 'Generating access logs...'
+    reply_stream_map = dict() # key: the tweet uid that is a reply, value: the target stream
+    user_timeline_map = dict() # key: the uuid of the stream, value: list containing the tweets' uuid for that stream
+    tweet_server = dict() # key: the tweet uuid, value: the corresponding server
+    tweet_uuid = dict()
+    with open(filename, 'rb') as dataset, open(filename + '.ready', 'wb') as result_file, open(UUID_LIST_FILE, 'wb') as uuid_file:
+        for line in dataset:
+            line_trimmed = line.strip()
+            line_splitted = line.split(' ')
+            timestamp = line_splitted[2]
+            mode = line_splitted[3].strip()
+            first_uid = line_splitted[0]
+            first_user_ip = user_ip_map[first_uid]
+            second_uid = line_splitted[1]
+            second_user_ip = user_ip_map[second_uid]
+            first_target_server = uuid_server_map[first_uid]
+            second_target_server = uuid_server_map[second_uid]
+            tweet_uid = str(uuid.uuid4())
+            if mode == TWITTER_MT:
+                triplet = (first_uid, second_uid, timestamp)
+                if triplet not in rt_re_set:  # make sure that we don't process the tweet twice
+                    # Three steps:
+                    #   1) A reads A's timeline
+                    #   2) A writes to his own timeline
+                    #   3) A reads his own timeline
+                    access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                    result_file.write(LOG_FORMAT % (timestamp, first_uid, first_user_ip, 'null', first_target_server, 'READ', '200', '2800'))
```

Comment:
this is redundant, right? isn't the `write` already called in access_timeline above?